src/localPRG.cpp

```diff
@@ -1183,62 +1184,110 @@ LocalPRG::find_alt_path(const std::vector<LocalNodePtr> &ref_path, const uint32_
-            added += kmer_path[1]->path.get_start() - n->pos.start;
+        }
+
+        bool local_node_starts_before_position = local_node->pos.start < position;
+        bool local_node_ends_before_position =
+                local_node->pos.get_end() <= position;
+        if (local_node_ends_before_position) {
+            number_of_bases_in_local_path_before_the_position += local_node->pos.length;
+        } else if (local_node_starts_before_position) {
+            number_of_bases_in_local_path_before_the_position += position - local_node->pos.start;
             break;
         }
     }
 
-    //std::cout << "added " << added << std::endl;
-    KmerNodePtr prev = nullptr;
-    for (const auto &n: kmer_path) {
-        //std::cout << n->path << " ";
-        if (n->path.length() == 0)
+    return number_of_bases_in_local_path_before_the_position;
+}
+
+uint32_t
+LocalPRG::get_number_of_bases_that_are_exclusively_in_the_previous_kmer_node(const KmerNodePtr &previous_kmer_node,
+                                                                             const KmerNodePtr &current_kmer_node) const {
+    uint32_t number_of_bases_that_are_exclusively_in_the_previous_kmer_node = 0;
+    auto previous_kmer_node_path_interval_iterator = previous_kmer_node->path.begin();
+
+    bool previous_kmer_node_path_interval_ends_before_current_kmer_node =
+            previous_kmer_node_path_interval_iterator->get_end() <=
+            current_kmer_node->path.get_start();
+    while (previous_kmer_node_path_interval_ends_before_current_kmer_node) {
+        number_of_bases_that_are_exclusively_in_the_previous_kmer_node += previous_kmer_node_path_interval_iterator->length;
+        previous_kmer_node_path_interval_iterator++;
+        previous_kmer_node_path_interval_ends_before_current_kmer_node =
+                previous_kmer_node_path_interval_iterator->get_end() <=
+                current_kmer_node->path.get_start();
+    }
+
+    if (current_kmer_node->path.get_start() > previous_kmer_node_path_interval_iterator->start) {
+        number_of_bases_that_are_exclusively_in_the_previous_kmer_node +=
+                current_kmer_node->path.get_start() - previous_kmer_node_path_interval_iterator->start;
+    }
+
+    return number_of_bases_that_are_exclusively_in_the_previous_kmer_node;
+}
+
+
+std::pair<std::vector<uint32_t>, std::vector<uint32_t>>
+LocalPRG::append_kmer_covgs_in_range(const KmerGraphWithCoverage &kmer_graph_with_coverage,
+                                     const std::vector<KmerNodePtr> &kmer_path,
+                                     const std::vector<LocalNodePtr> &local_path, const uint32_t &range_pos_start,
+                                     const uint32_t &range_pos_end, const uint32_t &sample_id) const {
+    assert(kmer_path.size() >
+           1); // this is an assert because it is the programmers responsibility to ensure that the kmer_path given to this function has at least size 1
```

Comment:
I think formatting moved the 1 to a new line.

Reply:
Yeah, I tried moving the comments out to check if it would format the 1 back to the previous line, but no success